### DISCUSSION TITLE
Miller 6

### DISCUSSION THREAD

**Discussion by johnkerl** [johnkerl/miller#851] [Announcements]
Please see [What's new in Miller 6](https://miller.readthedocs.io/en/latest/new-in-miller-6) for complete information.

<hr /><em>This discussion was created from the release <a href='https://github.com/johnkerl/miller/releases/tag/v6.0.0'>Miller 6</a>.</em>

**Comment by johnkerl** [comment]
@jauderho re #821 and #822 -- https://github.com/johnkerl/miller/runs/4756000917?check_suite_focus=true looks great!! :)

**Reply by johnkerl** [reply]
@jauderho correction, the `.tar.gz` files contain an executable named `miller` not `mlr`; currently debugging to see why ...

**Reply by johnkerl** [reply]
@jauderho this is resolved.

For posterity, what I did to trigger a re-run was:

* https://github.com/johnkerl/miller/pull/852
* Made a temp release https://github.com/johnkerl/miller/releases/tag/v6.0.0-goreleaserfix (which I afterward deleted)
* Download the `.tar.gz` files to local disk
* `tar ztf` / `tar zxf` the one for darwin amd64 (which I'm on) & verify `mlr` existed & was executable
* Rename all the `.tar.gz` files to have the filenames at https://github.com/johnkerl/miller/releases/tag/v6.0.0
* Update the checksums txt file with the correct names, and verified `shasum -a 256` was unchanged
* Went back to https://github.com/johnkerl/miller/releases/tag/v6.0.0
* Unattached the previous `.tar.gz` and checksums txt files
* Uploaded the new ones from local disk

Thanks again for setting this up BTW!! I regret only that I didn't catch the `miller` / `mlr` issue sooner on 6.0.0.rc1; I didn't look carefully. Besides that though -- this go-releaser is a fantastic feature! :D

**Reply by jauderho** [reply]
@johnkerl Congrats on the release. I'll have to go update my miller Docker builds to track the release instead of just git now. 

Your fix for re-releasing is a lot more goreleaser than I have previously worked with before. Definitely good to know in case I have to trigger a re-release for something else later.

I'm not 100% sure but setting `draft: true` in the `release` section of the yaml might help with the re-release. See https://goreleaser.com/customization/release/

**Reply by jauderho** [reply]
@johnkerl I've figured out how to use goreleaser to create Windows zip files so that can be part of the goreleaser process. Similarly, I've also figured out how to include deb/rpm builds. I am doing more testing but will submit a new pull request once I'm satisfied. 

Preview is here: https://github.com/jauderho/bl3auto/blob/main/.goreleaser.yml

**Reply by johnkerl** [reply]
@jauderho awesome!! :)

**Reply by jauderho** [reply]
See PR #866 and #867

**Comment by BEFH** [comment]
@johnkerl, should conda compile from https://github.com/johnkerl/miller/archive/refs/tags/v6.0.0.tar.gz or https://github.com/johnkerl/miller/releases/download/v6.0.0/miller-6.0.0.tar.gz? Traditionally I've used the tag (as did the original maintainers), but it looks like the files are different.

**Reply by johnkerl** [reply]
@BEFH the first one is auto-generated by GitHub & the other one is created by https://github.com/johnkerl/miller/blob/main/create-release-tarball. The first one is a tar of the entire git repo; the second one excludes some unnecessaries (https://github.com/johnkerl/miller/blob/main/create-release-tarball#L75-L81) but either one should have the necessaries to build with.

**Reply by BEFH** [reply]
It seems to have worked on MacOS, and it passed all of the unit tests on linux, but then when I ran `mlr --version`, it now segfaults. Also does it just running `mlr`.  I'm trying to rebuild now.

```
Reading symbols from mlr...
[New LWP 79998]
Core was generated by `./mlr --version'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fce97fff0a0 in ?? ()
```

**Reply by BEFH** [reply]
False alarm; it worked the second time.

**Reply by BEFH** [reply]
I've initiated the pull request and am waiting for CI checks to run on Azure. I will merve when and if they complete.

**Reply by BEFH** [reply]
OK, it segfaulted on the linux build check, just like the first time I ran it locally. Thoughts?

https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/438468/logs/10

**Reply by johnkerl** [reply]
@BEFH from
https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/438468/logs/10
I'm seeing a lot of `Unknown format` for `lib{varies}.a` -- maybe an unclean cross-compile with remnant artifacts from building for another architecture -- ?

**Comment by BEFH** [comment]
I don't know how that would happen; Azure is creating a brand-new clean
environment for each build using Docker.

It might be packages, even though the requirements are the same:

This worked on my test setup:

```
_go_select:              2.2.0-nocgo        conda-forge
_libgcc_mutex:           0.1-conda_forge    conda-forge
_openmp_mutex:           4.5-1_gnu          conda-forge
binutils_impl_linux-64:  2.36.1-h193b22a_2  conda-forge
binutils_linux-64:       2.36-hf3e587d_4    conda-forge
gcc_impl_linux-64:       11.2.0-h82a94d6_11 conda-forge
gcc_linux-64:            11.2.0-h39a9532_4  conda-forge
go:                      1.16.10-he1b62bb_0 conda-forge
kernel-headers_linux-64: 2.6.32-he073ed8_15 conda-forge
ld_impl_linux-64:        2.36.1-hea4e1c9_2  conda-forge
libgcc-devel_linux-64:   11.2.0-h0952999_11 conda-forge
libgcc-ng:               11.2.0-h1d223b6_11 conda-forge
libgomp:                 11.2.0-h1d223b6_11 conda-forge
libsanitizer:            11.2.0-he4da1e4_11 conda-forge
libstdcxx-ng:            11.2.0-he4da1e4_11 conda-forge
make:                    4.3-hd18ef5c_1     conda-forge
sysroot_linux-64:        2.12-he073ed8_15   conda-forge
```

And this failed on azure:

```
_go_select:              2.3.0-cgo          conda-forge
_libgcc_mutex:           0.1-conda_forge    conda-forge
_openmp_mutex:           4.5-1_gnu          conda-forge
binutils_impl_linux-64:  2.36.1-h193b22a_2  conda-forge
binutils_linux-64:       2.36-hf3e587d_4    conda-forge
gcc_impl_linux-64:       9.4.0-h03d3576_11  conda-forge
gcc_linux-64:            9.4.0-h391b98a_4   conda-forge
go:                      1.16.10-h17d2266_0 conda-forge
kernel-headers_linux-64: 2.6.32-he073ed8_15 conda-forge
ld_impl_linux-64:        2.36.1-hea4e1c9_2  conda-forge
libgcc-devel_linux-64:   9.4.0-hd854feb_11  conda-forge
libgcc-ng:               11.2.0-h1d223b6_11 conda-forge
libgfortran-ng:          11.2.0-h69a702a_11 conda-forge
libgfortran5:            11.2.0-h5c6108e_11 conda-forge
libgomp:                 11.2.0-h1d223b6_11 conda-forge
libsanitizer:            9.4.0-h79bfe98_11  conda-forge
libstdcxx-ng:            11.2.0-he4da1e4_11 conda-forge
make:                    4.3-hd18ef5c_1     conda-forge
sysroot_linux-64:        2.12-he073ed8_15   conda-forge
```

I just don't understand how it compiles and passes all the tests, but when
it's copied and tested a second time, it segfaults.

**Reply by johnkerl** [reply]
:(

**Reply by BEFH** [reply]
maybe I need to `set CGO_ENABLED=0`?

**Reply by johnkerl** [reply]
@BEFH worth a try -- ?

**Reply by johnkerl** [reply]
@BEFH I will need to do a local Conda build like we did before to help debug this

**Reply by BEFH** [reply]
Give me a little while longer on this; I'm trying to work it out now. The version that works on my computer is failing on conda-forge, and the version that works on conda-forge is failing on my computer, so I am trying to figure it out.

**Reply by BEFH** [reply]
OK, `go-nocgo` seems to have fixed things so they work on both Azure and my computer 🎉 . Anaconda is now up to date.

https://anaconda.org/conda-forge/miller

**Reply by johnkerl** [reply]
Awesome @BEFH !! :D

**Comment by herbygillot** [comment]
Miller 6 has been made available in MacPorts: https://ports.macports.org/port/miller/

**Reply by johnkerl** [reply]
Thanks @herbygillot !! :D

**Comment by jauderho** [comment]
My miller docker images are available here: https://hub.docker.com/r/jauderho/miller/tags

**Reply by johnkerl** [reply]
@jauderho this is awesome!! I'll connect this to the README / release info to help socialize it

**Reply by jauderho** [reply]
Example invocation: `docker run --rm -i jauderho/miller:latest --csv sort -f shape < ./example.csv` 

I also have an hourly GH action tracking for new release tags. So these images will be updated shortly after you tag any new releases.

**Reply by johnkerl** [reply]
Added to https://miller.readthedocs.io/en/latest/installing-miller/ -- thanks @jauderho ! :)
